File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        max_val = -1

        prev = ""
        cnt = 1

        def get_3_digit(n):
            i = 1
            res = n
            while i < 3:
                res *= 10
                res += n
                i += 1
            return res

        for ch in num:
            if ch == prev:
                cnt += 1
            else:
                cnt = 1
            
            if cnt == 3:
                curr = get_3_digit(int(ch))
                if curr > max_val:
                    max_res = ch * 3
                    max_val = curr
                print(ch)
                cnt = 1
            prev = ch

        if max_val == -1:
            return ""

        return max_res

File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.cs
public class Solution {
    public string LargestGoodInteger(string num) {
        int sameCount = 0;
        int high = -1;
        char previous = '0';
        
        foreach (char no in num){
            if(previous == no){
                sameCount++;
                
            }
            else{
                sameCount=0;
            }
            if(sameCount == 2){
               int current = no -'0';
                if(current>high){
                    high = current;
                }
            }
            previous = no;
        }
        if (high == -1)
            return "";
        int numb = high;
        numb = numb * 100 + numb *10 + numb* 1;
        
        if(numb == 0){
            return "000";
        }
         return numb.ToString();
    }
}